### PR TITLE
updated publication display_title to use ID rather than article title

### PR DESCRIPTION
**Pull request by aschroed**
The display_title now add the string in the ID field - either PMIDnnnnn or doi:10.... to the end of the short attribution instead of a (possibly) truncated version of the title.

This should be merged into or when the UI change to include the article title as the subheading of the Item page is made. 

**Update 06/29/2020:**

### CODE REVIEW

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404&bid=17741278)

File: src/encoded/static/components/item-pages/PublicationView.js [hunk 1]
```diff
@@ -7,11 +7,14 @@ import _ from 'underscore';
 import { console, object } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
 import { formatPublicationDate } from '@hms-dbmi-bgm/shared-portal-components/es/components/ui/LocalizedTime';
 
+import { PageTitleContainer, TitleAndSubtitleUnder, OnlyTitle, StaticPageBreadcrumbs, pageTitleViews } from './../PageTitle';
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/PublicationView.js:10]: Codacy found an issue: ['TitleAndSubtitleUnder' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404&bid=17741278)

File: src/encoded/static/components/item-pages/PublicationView.js [hunk 2]
```diff
@@ -7,11 +7,14 @@ import _ from 'underscore';
...
+import { Wrapper as ItemHeaderWrapper, TopRow, MiddleRow, BottomRow } from './components/ItemHeader';
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/PublicationView.js:11]: Codacy found an issue: ['ItemHeaderWrapper' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/DefaultItemView.js [hunk 3]
```diff
@@ -281,8 +282,9 @@ export default class DefaultItemView extends React.PureComponent {
      * @todo Maybe simplify CSS styling around these. Or get rid of these components and use plain HTML elements.
      */
     itemHeader(){
+        const { context, href, schemas, windoWidth } = this.props;
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/DefaultItemView.js:285]: Codacy found an issue: ['windoWidth' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 17
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/static/components/item-pages/components/Publications.js  2
         

Complexity decreasing per file
==============================
+ src/encoded/static/components/item-pages/components/AttributionTabView.js  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404&bid=17741278)

File: src/encoded/static/components/item-pages/PublicationView.js (hunk 1 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/PublicationView.js:10]: Codacy found an issue: ['TitleAndSubtitleUnder' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/PublicationView.js (hunk 2 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/PublicationView.js:11]: Codacy found an issue: ['ItemHeaderWrapper' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/DefaultItemView.js (hunk 3 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/DefaultItemView.js:285]: Codacy found an issue: ['windoWidth' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/components/AttributionTabView.js [hunk 4]
```diff
@@ -163,20 +163,26 @@ class LabsSection extends React.PureComponent {
...
+        const { mounted } = this.state;
+
+        if (!primaryLab && contribLabLen === 0) return null;
 
-        if (!primaryLab && !contributingLabs) return null;
         return (
             <div className={className}>
-                { primaryLab ? FormattedInfoBlock.Lab(primaryLab, true, true, isMounted) : null }
-                { contributingLabs ?
+                { primaryLab ? FormattedInfoBlock.Lab(primaryLab, true, true, mounted) : null }
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/AttributionTabView.js:180]: Codacy found an issue: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 17
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/static/components/item-pages/components/Publications.js  6
         

Complexity decreasing per file
==============================
+ src/encoded/static/components/item-pages/components/AttributionTabView.js  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404&bid=17741278)

File: src/encoded/static/components/item-pages/PublicationView.js (hunk 1 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/PublicationView.js:10]: Codacy found an issue: ['TitleAndSubtitleUnder' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/PublicationView.js (hunk 2 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/PublicationView.js:11]: Codacy found an issue: ['ItemHeaderWrapper' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/DefaultItemView.js (hunk 3 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/DefaultItemView.js:285]: Codacy found an issue: ['windoWidth' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/components/AttributionTabView.js (hunk 4 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/AttributionTabView.js:180]: Codacy found an issue: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/components/Publications.js [hunk 5]
```diff
@@ -206,6 +209,19 @@ const PublicationDetailRows = React.memo(function PublicationDetailRows({ public
         );
     }
 
+    if (externalID && typeof externalID === 'string' && title){
+        // Exclude if no `title`, since ID is present in `display_title` already.
```
Comment (alexkb0009) [inline src/encoded/static/components/item-pages/components/Publications.js:213]: Maybe worth changing later. Am assuming 99%+ pubs will have ID (?)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 17
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/static/components/item-pages/components/Publications.js  6
         

Complexity decreasing per file
==============================
+ src/encoded/static/components/item-pages/components/AttributionTabView.js  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404&bid=17741278)

File: src/encoded/static/components/item-pages/PublicationView.js (hunk 1 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/PublicationView.js:10]: Codacy found an issue: ['TitleAndSubtitleUnder' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/PublicationView.js (hunk 2 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/PublicationView.js:11]: Codacy found an issue: ['ItemHeaderWrapper' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/DefaultItemView.js (hunk 3 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/DefaultItemView.js:285]: Codacy found an issue: ['windoWidth' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/components/AttributionTabView.js [hunk 6]
```diff
@@ -164,20 +164,26 @@ class LabsSection extends React.PureComponent {
...
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/AttributionTabView.js:181]: Codacy found an issue: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 17
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/static/components/item-pages/components/Publications.js  6
         

Complexity decreasing per file
==============================
+ src/encoded/static/components/item-pages/components/AttributionTabView.js  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404&bid=17741278)

File: src/encoded/static/components/item-pages/DefaultItemView.js (hunk 3 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/DefaultItemView.js:285]: Codacy found an issue: ['windoWidth' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/components/AttributionTabView.js (hunk 6 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/AttributionTabView.js:181]: Codacy found an issue: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/PublicationView.js (hunk 1 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/PublicationView.js:10]: Codacy found an issue: ['TitleAndSubtitleUnder' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

File: src/encoded/static/components/item-pages/PublicationView.js (hunk 2 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/PublicationView.js:11]: Codacy found an issue: ['ItemHeaderWrapper' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5411404)

**Pull request merged**